### PR TITLE
FIX: Error when trying to bump a topic with no category

### PR DESCRIPTION
**Pull request by janzenisaac**
When revising a post, if the topic that post belonged to did not have a category attached it would error with 

> NoMethodError (undefined method `read_restricted' for nil:NilClass)

### CODE REVIEW

File: spec/lib/post_revisor_spec.rb [hunk 1]
```diff
@@ -422,6 +422,20 @@
...
+        post_from_topic_with_no_category = Fabricate(:post, topic: topic_with_no_category)
+        expect {
+          result =
+            subject.revise!(
+              Fabricate(:admin),
+              raw: post_from_topic_with_no_category.raw,
+              tags: ["foo"],
+            )
+          expect(result).to eq(true)
+        }.to change { topic.reload.bumped_at }
```
Comment (davidtaylorhq) [inline spec/lib/post_revisor_spec.rb:436]: The test description says "not bump topic"... but here we're checking that it was bumped 🤔

File: app/models/topic_tracking_state.rb [hunk 2]
```diff
@@ -579,7 +579,7 @@ def self.trigger_post_read_count_update(post, groups, last_read_post_number, use
   def self.secure_category_group_ids(topic)
     category = topic.category
 
-    if category.read_restricted
+    if category&.read_restricted
```
Comment (davidtaylorhq) [inline app/models/topic_tracking_state.rb:582]: This method is responsible for returning the group_ids for the message-bus notification. I think returning "nil" when the category is nil means the message-bus notification will be public.

How about returning only the admin group if category is nil?

File: spec/lib/post_revisor_spec.rb (hunk 1 above)
Comment (SamSaffron) [reply spec/lib/post_revisor_spec.rb:436]: fiddling with topic tracking state feels risky. not even sure what a topic without a category is? even uncategorized/general gets a category id.

Comment (davidtaylorhq) [reply spec/lib/post_revisor_spec.rb:436]: Yeah, a topic without a category should not exist. But our lack of enforced foreign-keys makes it possible, so we need to be able to handle the situation without blowing up.

**Pull request merged**